src/Sulu/Component/Content/Tests/Unit/Document/Subscriber/PHPCR/SuluNodeTest.php

```diff
@@ -24,74 +25,544 @@ class SuluNodeTest extends TestCase
+        $node->getPrimaryNodeType()->willReturn($nodeType);
+
+        $suluNode = new SuluNode($node->reveal());
+
+        $this->assertSame($nodeType, $suluNode->getPrimaryNodeType());
+    }
+
+    public function testGetMixinNodeTypes(): void
+    {
+        $nodeTypes = [$this->prophesize(NodeTypeInterface::class)->reveal()];
+        $node = $this->prophesize(NodeInterface::class);
+        $node->getMixinNodeTypes()->willReturn($nodeTypes);
+
+        $suluNode = new SuluNode($node->reveal());
+
+        $this->assertSame($nodeTypes, $suluNode->getMixinNodeTypes());
+    }
+
+    public function testIsNodeType(): void
+    {
+        $node = $this->prophesize(NodeInterface::class);
+        $node->isNodeType('sulu:content')->willReturn(true);
+
+        $suluNode = new SuluNode($node->reveal());
+
+        $this->assertTrue($suluNode->isNodeType('sulu:content'));
+    }
+
+    public function testSetPrimaryType(): void
+    {
+        $node = $this->prophesize(NodeInterface::class);
+        $node->setPrimaryType('sulu:content')->shouldBeCalled();
+
+        $suluNode = new SuluNode($node->reveal());
+
+        $suluNode->setPrimaryType('sulu:content');
+    }
+
+    public function testAddMixin(): void
+    {
+        $node = $this->prophesize(NodeInterface::class);
+        $node->addMixin('sulu:content')->shouldBeCalled();
+
+        $suluNode = new SuluNode($node->reveal());
+
+        $suluNode->addMixin('sulu:content');
+    }
+
+    public function testRemoveMixin(): void
+    {
+        $node = $this->prophesize(NodeInterface::class);
+        $node->removeMixin('sulu:content')->shouldBeCalled();
+
+        $suluNode = new SuluNode($node->reveal());
+
+        $suluNode->removeMixin('sulu:content');
+    }
+
+    public function testSetMixins(): void
+    {
+        $node = $this->prophesize(NodeInterface::class);
+        $node->setMixins(['sulu:content'])->shouldBeCalled();
+
+        $suluNode = new SuluNode($node->reveal());
+
+        $suluNode->setMixins(['sulu:content']);
+    }
+
+    public function testUpdate(): void
+    {
+        $node = $this->prophesize(NodeInterface::class);
+        $node->update('default')->shouldBeCalled();
+
+        $suluNode = new SuluNode($node->reveal());
+
+        $suluNode->update('default');
+    }
+
+    public function testGetCorrespondingNodePath(): void
+    {
+        $node = $this->prophesize(NodeInterface::class);
+        $node->getCorrespondingNodePath('default')->willReturn('/test-1');
+
+        $suluNode = new SuluNode($node->reveal());
+
+        $this->assertSame('/test-1', $suluNode->getCorrespondingNodePath('default'));
+    }
+
+    public function testGetSharedSet(): void
+    {
+        $nodes = [];
+        $node = $this->prophesize(NodeInterface::class);
+        $node->getSharedSet()->willReturn($nodes);
+
+        $suluNode = new SuluNode($node->reveal());
+
+        $this->assertSame($nodes, iterator_to_array($suluNode->getSharedSet()));
```

Comment:
if you want to convert a `iterable` (array / Traversable) to an array you can use `[...$suluNode->getSharedSet()]` this works with arrays and traversables.